Source/Scene/PostProcessAmbientOcclusionStage.js

```diff
@@ -0,0 +1,199 @@
+define([
+        '../Core/Check',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/PixelFormat',
+        '../Renderer/PixelDatatype',
+        '../Renderer/Sampler',
+        '../Renderer/Texture',
+        '../Renderer/TextureMagnificationFilter',
+        '../Renderer/TextureMinificationFilter',
+        '../Renderer/TextureWrap',
+        '../Shaders/PostProcessFilters/AmbientOcclusion',
+        '../Shaders/PostProcessFilters/AmbientOcclusionGenerate',
+        './PostProcess',
+        './PostProcessBlurStage',
+        './PostProcessComposite'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        destroyObject,
+        PixelFormat,
+        PixelDatatype,
+        Sampler,
+        Texture,
+        TextureMagnificationFilter,
+        TextureMinificationFilter,
+        TextureWrap,
+        AmbientOcclusion,
+        AmbientOcclusionGenerate,
+        PostProcess,
+        PostProcessBlurStage,
+        PostProcessComposite) {
+    'use strict';
+
+    /**
+     * Post process stage for ambient occlusion. Implements {@link PostProcess}.
+     *
+     * @alias PostProcessAmbientOcclusionStage
+     * @constructor
+     *
+     * @private
+     */
+    function PostProcessAmbientOcclusionStage() {
+        this._name = 'czm_ambient_occlusion';
+        this._randomTexture = undefined;
+
+        var that = this;
+        this._generatePostProcess = new PostProcess({
+            name : 'czm_ambient_occlusion_generate',
+            fragmentShader : AmbientOcclusionGenerate,
+            uniformValues : {
+                intensity : 4.0,
+                bias : 0.0,
+                lengthCap : 0.25,
+                stepSize : 2.0,
+                frustumLength : 1000.0,
+                randomTexture : function () {
+                    return that._randomTexture;
+                }
+            }
+        });
+        this._blurPostProcess = new PostProcessBlurStage({
+            name : 'czm_ambient_occlusion_blur'
+        });
+        var generateComposite = new PostProcessComposite({
+            name : 'czm_ambient_occlusion_generate_blur',
+            processes : [this._generatePostProcess, this._blurPostProcess]
+        });
+
+        var ambientOcclusionComposite = new PostProcess({
+            name : 'czm_ambient_occlusion_composite',
+            fragmentShader : AmbientOcclusion,
+            uniformValues : {
+                ambientOcclusionOnly : false,
+                ambientOcclusionTexture : this._blurPostProcess.name
+            }
+        });
+
+        this._compositeProcess = new PostProcessComposite({
+            name : 'czm_ambient_occlusion_generate_and_composite',
+            processes : [generateComposite, ambientOcclusionComposite],
+            executeInSeries : false
```

Comment:
This parameter was confusing to me at first glance - wouldn't all stages happen in series?

Can this be removed and instead can we just have named inputs for what is the previous external stage / rendered scene vs. stage in the composite output?

Or maybe just rename this to make it clear that it impacts the input.